h2/src/main/org/h2/mvstore/MVMap.java

```diff
@@ -1305,24 +1361,25 @@ public void append(K key, V value) {
      * Non-updating method may be used concurrently, but latest removal may not be visible.
      */
     public void trimLast() {
-        int attempt = 0;
-        boolean success;
-        do {
-            RootReference rootReference = getRootInternal();
-            int appendCounter = rootReference.getAppendCounter();
-            if (appendCounter > 0) {
-                RootReference updatedRootReference = new RootReference(rootReference, appendCounter - 1, ++attempt);
-                success = root.compareAndSet(rootReference, updatedRootReference);
-            } else {
-                assert rootReference.root.getKeyCount() > 0;
-                Page lastLeaf = rootReference.root.getAppendCursorPos(null).page;
-                assert lastLeaf.isLeaf();
-                assert lastLeaf.getKeyCount() > 0;
-                Object key = lastLeaf.getKey(lastLeaf.getKeyCount() - 1);
-                success = remove(key) != null;
-                assert success;
+        RootReference rootReference = getRoot();
+        int appendCounter = rootReference.getAppendCounter();
+        boolean fallback = appendCounter == 0;
```

Comment:
It's not clear for me why such name was chosen, may be use a some better name or add a comment?